graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -185,37 +199,41 @@ class RenameSchemaTypesVisitor(Visitor):
             "UnionTypeDefinitionNode",
         }
     )
-
-    def __init__(self, renamings, query_type, scalar_types):
+    type_hint_rename_types = Union[
+        EnumTypeDefinitionNode,
+        InterfaceTypeDefinitionNode,
+        NamedTypeNode,
+        ObjectTypeDefinitionNode,
+        UnionTypeDefinitionNode,
+    ]
+
+    def __init__(self, renamings: Dict[str, str], query_type: str, scalar_types: Set[str]) -> None:
         """Create a visitor for renaming types in a schema AST.
 
         Args:
-            renamings: Dict[str, str], mapping from original type name to renamed type name.
-                       Any name not in the dict will be unchanged
-            query_type: str, name of the query type (e.g. RootSchemaQuery), which will not
-                        be renamed
-            scalar_types: Set[str], set of names of all scalars used in the schema, including
-                          all user defined scalars and any builtin scalars that were used
+            renamings: Maps original type name to renamed name. Any name not in the dict will be
+                       unchanged
+            query_type: Name of the query type (e.g. RootSchemaQuery), which will not be renamed
+            scalar_types: The set of all scalars used in the schema, including all user defined
+                          scalars and any builtin scalars that were used
         """
         self.renamings = renamings
-        self.reverse_name_map = {}  # Dict[str, str], from renamed type name to original type name
+        self.reverse_name_map: Dict[str, str] = {}  # from renamed type name to original type name
         # reverse_name_map contains all types, including those that were unchanged
         self.query_type = query_type
         self.scalar_types = frozenset(scalar_types)
         self.builtin_types = frozenset({"String", "Int", "Float", "Boolean", "ID"})
 
-    def _rename_name_and_add_to_record(self, node):
+    def _rename_name_and_add_to_record(self, node: type_hint_rename_types) -> Node:
```

Comment:
The `Returns` section of the docstring (`Node object, identical to the input node, except with possibly a new name`) seems to suggest a tighter type bound here:
```
NodeT = TypeVar("NodeT", bound=Node)
def _rename_name_and_add_to_record(self, node: NodeT) -> NodeT:
```

Thoughts?

Reply:
According to the bound documentation, it seems like this code just says the argument and return have to be a subclass of `Node` which doesn't seem to be a tighter bound. Depending on what we end up deciding on in the [discussion above](https://github.com/kensho-technologies/graphql-compiler/pull/847#discussion_r438832613), it looks the same as just using the type from there. Am I missing something here?

Reply:
We can tighten this by setting the return type to be `RenameTypes`. Unfortunately, the current linter version complains about this [and it was fixed recently](https://github.com/PyCQA/pyflakes/pull/448), so I'm currently looking into updating our dependencies. It seems like in the process of updating the Pipfile, we updated `pylint` and the new version can take a really long time to run.

I've committed the change here to see if linting works with Travis CI on Github-- if it does, I'll revert the change here and split the version bump into a separate PR.

Reply:
Turns out making it a module-level attribute did indeed work! Updating that now and then the PR will be good to go.

Reply:
Will update with generic typing--just read the discussion on the difference between this and the existing type-hint